network/client_test.go

```diff
@@ -24,6 +24,9 @@ func testClient(t *testing.T, fac routerFactory, cl clientFactory) {
 		t.Fatal(err)
 	}
 	go r.Start()
+	for !r.Listening() {
+		time.Sleep(20 * time.Millisecond)
+	}
```

Comment:
Yes, it could if something is wrong, and if so, it's a problem we should solve, so I'm happy that it's in the test ;) 
This line was added just to make sure the test was not failing because of the listener not really listening yet, but in fact it was not that. Still it's a good habit to make sure the listener is really listening before going on with our test. That's the same code which is in `sda.LocalTest` which is already merged...